pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -146,40 +147,56 @@ type PlayerSelector struct {
 }
 
 // CounterSelector is the filter options for a GameServer based on the count and/or available capacity.
-// 0 for MaxCount or MaxAvailable means unlimited maximum. Default for all fields: 0
 type CounterSelector struct {
-	MinCount     int64 `json:"minCount"`
-	MaxCount     int64 `json:"maxCount"`
+	// MinCount is the minimum current value. Defaults to 0.
+	// +optional
+	MinCount int64 `json:"minCount"`
+	// MaxCount is the maximum current value. Defaults to 0, which translates as max(in64).
+	// +optional
+	MaxCount int64 `json:"maxCount"`
+	// MinAvailable specifies the minimum capacity (current capacity - current count) available on a GameServer. Defaults to 0.
+	// +optional
 	MinAvailable int64 `json:"minAvailable"`
+	// MaxAvailable specifies the maximum capacity (current capacity - current count) available on a GameServer. Defaults to 0, which translates to max(int64).
+	// +optional
 	MaxAvailable int64 `json:"maxAvailable"`
 }
 
 // ListSelector is the filter options for a GameServer based on List available capacity and/or the
 // existence of a value in a List.
-// 0 for MaxAvailable means unlimited maximum. Default for integer fields: 0
-// "" for ContainsValue means ignore field. Default for string field: ""
 type ListSelector struct {
+	// ContainsValue says to only match GameServers who has this value in the list. Defaults to "", which is all.
+	// +optional
 	ContainsValue string `json:"containsValue"`
-	MinAvailable  int64  `json:"minAvailable"`
-	MaxAvailable  int64  `json:"maxAvailable"`
+	// MinAvailable specifies the minimum capacity (current capacity - current count) available on a GameServer. Defaults to 0.
+	// +optional
+	MinAvailable int64 `json:"minAvailable"`
+	// MaxAvailable specifies the maximum capacity (current capacity - current count) available on a GameServer. Defaults to 0, which is translated as max(int64).
+	// +optional
+	MaxAvailable int64 `json:"maxAvailable"`
 }
 
 // CounterAction is an optional action that can be performed on a Counter at allocation.
-// Action: "Increment" or "Decrement" the Counter's Count (optional). Must also define the Amount.
-// Amount: The amount to increment or decrement the Count (optional). Must be a positive integer.
-// Capacity: Update the maximum capacity of the Counter to this number (optional). Min 0, Max int64.
 type CounterAction struct {
-	Action   *string `json:"action,omitempty"`
-	Amount   *int64  `json:"amount,omitempty"`
-	Capacity *int64  `json:"capacity,omitempty"`
+	// Action must to either "Increment" or "Decrement" the Counter's Count. Must also define the Amount.
+	// +optional
+	Action *string `json:"action,omitempty"`
```

Comment:
(not for this release) default to increment as a quality of life improvement?

Reply:
If amount != nil default to decrement? We could do that.

Reply:
🤔 I also think ignore this suggestion as well - given the rest!